### PR TITLE
fix: reorganize role permissions

### PR DESCRIPTION
**Pull request by dpgaspar**
### SUMMARY

Reorganizes existing role permissions for Alpha, Gamma and SQLLab

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by betodealmeida — APPROVED** [review]
Looks good!

**Review by villebro — APPROVED** [review]
Looks great!

**Comment by sadpandajoe** [comment]
🏷️ preset:2023.7

File: superset/security/manager.py [hunk 1]
```diff
@@ -154,40 +154,53 @@ class SupersetSecurityManager(  # pylint: disable=too-many-public-methods
     READ_ONLY_MODEL_VIEWS = {"Database", "DruidClusterModelView", "DynamicPlugin"}
 
     USER_MODEL_VIEWS = {
+        "RegisterUserModelView",
```
Comment (hash-data) [inline superset/security/manager.py:157]: hi @dpgaspar is their any use case to add it. Can you elaborate if their is ?

**Pull request merged**